subprojects/functional/src/main/java/org/gradle/internal/lazy/LockingLazy.java

```diff
@@ -50,35 +53,19 @@ public LockingLazy(Supplier<T> supplier) {
 
     @Override
     public T get() {
-        return ensureValue();
-    }
-
-    private T ensureValue() {
```

Comment:
I actually tried to break this by running with a varying number of concurrent threads and couldn't get it to fail with Java 11 on macOS.  

It's possible this was good enough or only caused concurrency problems in specific combinations of OS/JVM/thread contention.

I trust that Guava's implementation is probably the best we could do, so I switched to that here.